android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -531,14 +553,17 @@ public void mediaStreamRelease(final String id) {
 
             localStreams.remove(id);
 
-            // XXX MediaStream.dispose() cannot be invoked (on stream) because
-            // we do not know whether stream is still added to a PeerConnection.
-            // If it is, then MediaStream.dispose() will assert because it will
-            // expect zero references to stream but stream will have a positive
-            // number of references. Generally, PeerConnection.dispose() will
-            // dispose of the MediaStreams added to the PeerConnection but,
-            // unfortunately, we do not dispose of the PeerConnection at the
-            // time of this writing.
+            // MediaStream dispose() can be only called if it does no longer
+            // belong to any PeerConnection or the app wil crash (MediaStream
+            // counts references to it and throws an exception if disposed with
+            // any active references).
+            List<PeerConnectionObserver> pcs
+                = peerConnectionsForLocalStream(stream);
```

Comment:
We don't need `peerConnectionsForLocalStream`. There's no point in building a list of `PeerConnectionObserver`s which contain a specific `MediaStream` and then going through the built list and telling these to remove the `MediaStream`. That's an unnecessary `List` building and two searches in `PeerConnectionObserver` for a single `MediaStream`. What `WebRTCModule` needs is to tell every `PeerConnectionObserver` to remove a specific `MediaStream` because it's going to dispose the `MediaStream`. Whether a `PeerConenctionObserver` contains or does not contain the actual `MediaStream` is irrelevant - it will successfully remove the `MediaStream` in both cases anyway.

This will make the methods `peerConnectionsForLocalStream`  and `containsLocalStream` obsolete.